src/language/visitor.ts

```diff
@@ -17,34 +18,34 @@ export type Visitor<KindToNode, Nodes = $Values<KindToNode>> =
       KindToNode,
       <Node>(Node) => VisitFn<Nodes, Node> | EnterLeave<VisitFn<Nodes, Node>>,
     >;
-type EnterLeave<T> = {| +enter?: T, +leave?: T |};
+type EnterLeave<T> = { readonly enter?: T, readonly leave?: T };
 type ShapeMap<O, F> = $Shape<$ObjMap<O, F>>;
```

Comment:
@IvanGoncharov I am confused on how to convert this`$ObjMap`?

Reply:
Do you mean this? https://github.com/Microsoft/TypeScript/issues/20352

Reply:
Yes but I think there is a utility function for that in here https://github.com/graphql/graphql-js/blob/16.x.x/src/jsutils/ObjMap.js . I did try using that but it didn’t make sense  so I  am little confused

Reply:
It's seems to have one type parameter less, it might work better with de declaration from the linked issue. But need to dig deeper to investigate. Will try tomorrow

Reply:
how do you validate the syntax? I tried runnning `npx typescript` in the checkout of your branch, but to gave a lot (confusing) output

Reply:
yes this is still work in progress and there are many issues that need to be resolved

Reply:
@saihaj It's two different things:
* `$ObjMap` is Flow utility function https://flow.org/en/docs/types/utilities/#toc-objmap
* `ObjMap` is our utility type: https://github.com/graphql/graphql-js/blob/16.x.x/src/jsutils/ObjMap.js

Reply:
So  please replace `$ObjMap` with TS equvivalent

Reply:
> Do you mean this? [microsoft/TypeScript#20352](https://github.com/microsoft/TypeScript/issues/20352)

@IvanGoncharov so as @joostheijkoop mentioned that is the `$ObjMap` is TS equivalent. My questions are:
* Does this utility go in `src/jsutils`
  * Does this need any tests?
* What do I even name this utility since `ObjMap` is already a utility in codebase?
* Or should this change be handled in a new PR once we move `16.x.x` to `main`? all this conversation can be moved to an issue which can be handled by me or someone else in new PR?